simpleflow/swf/executor.py

```diff
@@ -653,3 +652,24 @@ def fail(self, reason, details=None):
 
     def run(self, decision_response):
         return self.replay(decision_response)
+
+    def get_execution_context(self):
+        return self._execution_context
+
+    def build_execution_context(self, decision_response):
+        """
+        Extract data from the execution and history.
+        :param decision_response:
+        :type  decision_response: swf.responses.Response
+        """
+        execution = decision_response.execution
+        history = decision_response.history
+        workflow_started_event = history[0]
+        # The "if execution else None" are for tests that don't provide an execution object
+        self._execution_context = dict(
+            name=execution.workflow_type.name if execution else None,
+            version=execution.workflow_type.version if execution else None,
+            workflow_id=execution.workflow_id if execution else None,
+            run_id=execution.run_id if execution else None,
+            tag_list=getattr(workflow_started_event, 'tag_list', []),
```

Comment:
Overly defensive? Do you think tag_list could be None here? I tried to read the code but the history parsing is sooo complicated that I'm not sure anymore now.